docs/DeveloperGuide.md

```diff
@@ -1,34 +1,234 @@
+- `Logic` : The Logic component makes sense of the command and executes it.
+- `Model` : The Model component is responsible for all data held in the memory.
+- `Storage` : The Storage component handles data by reading from and writing to files in the hard disk. 
+
+The sections below give more details of each component.
+
+#### 2.1.1 Ui Component
+The Ui component is responsible for getting the user input and for displaying appropriate messages as response 
+to commands executed by the logic component. The following class diagram gives a brief overview of the Ui component 
+and it's interaction with other components.
+
+![uicomponent](DG_Diagrams/UiComponent.png)
+
+#### 2.1.2. Logic Component
+The Logic component is responsible for the following tasks:
+- Makes sense of the command entered by the user by splitting it into the command type and the parameters and/or 
+delimiters if any.
+- Executes a command based on its type and provides an appropriate result to the user via the Ui component.
+
+The following class diagram briefly explains how different classes in the Logic component interact with each other.
+
+<img src="DG_Diagrams/LogicComponent.png" alt="logiccomponent" width=1100 height=500>
+
+#### 2.1.3. Model Component
+The Model component is responsible for the following tasks:
+- Stores the bus data, dine info data and favourites list in memory.
+- Performs operations on the data to facilitate the command execution by Logic component.
+
+The following class diagram briefly explains how different classes in the Model component interact with each other.
+
+![modelcomponet](DG_Diagrams/ModelComponent.png)
+
+#### 2.1.4. Storage Component
+The Storage component is responsible for the following tasks:
+- Stores the favourite commands in a file in a certain format and retrieves this information everytime 
+the user enters the application.
+- Stores the number of times each bus stop has been typed by the user since he/she started using the application and 
+retrieves it everytime the user enters the application.
+    + This information is used to find out the user's most frequently searched location so far and display it to the 
+    user. 
+- These files are updated after every execution of a user command.
+
+The following class diagram briefly explains how different classes in the Storage component interact with each other.
+
+![StorageComponent](DG_Diagrams/StorageComponent.png)
+
+### 2.2. Implementation
+This section provides details of how the main features of Nav@NUS have been implemented.
+
+#### 2.2.1 Direct Route Finder (`/route` Feature)
+
+`/route <location1> /to <location2>` is the command that has to entered by the user to see all direct bus routes 
+available from **location1** to **location2**.
+
+The class diagram in the figure below shows how different classes used for implementation of the `/route` command 
+are linked to each other. 
+
+![RouteCommandClass](DG_Diagrams/RouteCommandClass.png)
```

Comment:
The empty dash arrow might need <<interface>> in the box its pointing to? The rest of it looks good